tests/compatibility_tests/compatibility_test_runner.py

```diff
@@ -136,6 +123,7 @@ def construct_pytest_arguments(version_to_test_against):
     return pytest_args
 
 def run_test_mode(version_to_test_against):
+    logger.info(f"Beginning test mode on all test cases for version: {version_to_test_against}")
```

Comment:
I think this log line is a bit misleading, should say something like Beginning test mode on all test cases marked with versions less than or equal to version_to_test_against. 
Alternatively, why not have something like logger.info(f"===================================== TEST ODE RUN BEGINS =================================================")